### PR TITLE
Add watchWithVersion interface for etcdKV

### LINKED ISSUE #6629: specify the parameters for the etcd version when using etcdKV.WatchWithPrefix()
#### What would you like to be added:
add a ```WatchWithPrefix(key string, version int64) clientv3.WatchChan``` for etcdKV

### PR DESCRIPTION
**Pull request by xiaocai2333**
Signed-off-by: xiaocai2333 <cai.zhang@zilliz.com>

fix issue: #6629

### CODE REVIEW

File: internal/kv/etcd/etcd_kv_test.go [hunk 1]
```diff
@@ -308,6 +309,33 @@ func TestEtcdKV_WatchPrefix(t *testing.T) {
...
+
+	defer etcdKV.Close()
+	defer etcdKV.RemoveWithPrefix("")
+
+	key := "test-version"
+
+	err = etcdKV.Save(key, "v")
+	assert.Nil(t, err)
+
+	resp, _ := cli.Get(context.Background(), rootPath+"/"+key, clientv3.WithPrefix())
```
Comment (xige-16) [inline internal/kv/etcd/etcd_kv_test.go:326]: why not provide an interface to get max version from etcd?

**Comment by xiaocai2333** [comment]
fix issue: #6629

**Pull request merged**